base/server/share/lib/systemd/system/pki-tomcatd-nuxwdog@.service

```diff
@@ -14,7 +14,7 @@ EnvironmentFile=-/etc/sysconfig/%i
 ExecStartPre=+/usr/bin/setfacl -m u:pkiuser:wx /run/systemd/ask-password
 ExecStartPre=/usr/bin/pki-server-nuxwdog
 ExecStartPre=/usr/sbin/pki-server upgrade --validate %i
-ExecStartPre=/usr/sbin/pki-server migrate %i
+ExecStartPre=!/usr/sbin/pki-server migrate %i
```

Comment:
What does the `!` here mean? If you could point to the doc that'd be great. Thanks!

Reply:
See "Table 1. Special executable prefixes": https://www.freedesktop.org/software/systemd/man/systemd.service.html

Reply:
@cipherboy, thank you.

Reply:
The process is executed with full privileged, see https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStart=

Reply:
Unfortunately, root is required since NSS upgrade operation doesn't respect original permissions. For example, `certmonger` triggers such procedure as it is shown in
https://pagure.io/dogtagpki/issue/3104

Reply:
IMO this is a design bug in Dogtag. Dogtag should never rely on DAC override privileges and always execute certutil and similar commands as ``${pki_user}`` and group ``${pki_group}``. This would also solve my problems with SoftHSM2 integration.

Reply:
Thanks for the info! I'm OK with this particular change actually. Currently we run some of the upgrade operations during RPM installation because it needs root privileges, which might be more difficult to detect & troubleshoot if there's a problem. If we can move all upgrade operations into server startup it will make things easier.

Reply:
DAC override means that a user is allowed to access files although discrete access control (user, group, other privileges) does not allow the user to access files or directories. The root user does not have DAC permission to access /etc/pki/pki-tomcat/alias as the files are owned by pkiuser:pkiuser and have no other read/write/exec bit.

Reply:
I agree the design needs to be fixed, but that might take some time. If this particular issue is urgent, I don't mind adding `!` as a temporary solution until we fix the design.